ddht/v5_1/alexandria/client.py

```diff
@@ -110,6 +131,93 @@ async def _request(
             )
         return response  # type: ignore
 
+    def subscribe(
+        self,
+        message_type: Type[TAlexandriaMessage],
+        endpoint: Optional[Endpoint] = None,
+        node_id: Optional[NodeID] = None,
+    ) -> AsyncContextManager[
+        trio.abc.ReceiveChannel[InboundMessage[TAlexandriaMessage]]
+    ]:
+        return self.subscription_manager.subscribe(message_type, endpoint, node_id,)  # type: ignore
+
+    @asynccontextmanager
+    async def subscribe_request(
+        self,
+        node_id: NodeID,
+        endpoint: Endpoint,
+        request: AlexandriaMessage[Any],
+        response_message_type: Type[TAlexandriaMessage],
+        request_id: Optional[bytes],
+    ) -> AsyncIterator[trio.abc.ReceiveChannel[InboundMessage[TAlexandriaMessage]]]:
+        if request_id is None:
```

Comment:
I think there might be a race condition if we ever add an `await` between the call to `get_free_request_id` and the call to `reserve_request_id`. reordering some of these blocks such that one happens right after the other means there's a slightly lower chance someone will add that `await` in the future.

Reply:
I'll add a comment to document this subtlty.